request/multiple_kernels.rst

```diff
@@ -0,0 +1,183 @@
+.. index::
+   single: kernel, performance
+
+How To Create Symfony Applications with Multiple Kernels
+========================================================
+
+In most Symfony applications, incoming requests are processed by the
+``web/app.php`` front controller, which instantiates the ``app/AppKernel.php``
+class to create the application kernel that loads the bundles and handles the
+request to generate the response.
+
+This single kernel approach is a convenient default provided by the Symfony
+Standard edition, but Symfony applications can define any number of kernels.
+Whereas :doc:`environments </configuration/environments>` execute the same
+application with different configurations, kernels can execute different parts
+of the same application.
+
+These are some of the common use cases for creating multiple kernels:
+
+* An application that defines an API could define two kernels for performance
+  reasons. The first kernel would serve the regular application and the second
+  one would only respond to the API requests, loading less bundles and enabling
+  less features;
+* A highly sensitive application could define two kernels. The first one would
+  only load the routes that match the parts of the application exposed publicly.
+  The second kernel would load the rest of the application and its access would
+  be protected by the web server;
+* A micro-services oriented application could define several kernels to
+  enable/disable services selectively turning a traditional monolith application
+  into several micro-applications.
+
+Adding a new Kernel to the Application
+--------------------------------------
+
+Creating a new kernel in a Symfony application is a three-step process:
+
+1. Create a new front controller to load the new kernel;
```

Comment:
There is another way to do this (which is done on a BIG Symfony site I know of): keeping `app.php` and `app_dev.php` and using some URL matching to load different kernels. For example, if the URL starts with `/admin`, then load the `AdminKernel`, else `AppKernel`. I think it's worth mentioning both. So, what about:

A) Making step (1) actually step (3) (I think creating the kernel and hooking up the configuration makes sense to have first, then finally show how you can instantiate the now-created kernel)

B) Rename this step to `Step 3) Executing the kernel from a front controller`

C) Mention the `if` statement approach above (it could be mentioned first or second).